org.eclipse.scanning.api/src/org/eclipse/scanning/api/event/core/ResponseConfiguration.java

```diff
@@ -63,22 +65,22 @@ public void setTimeUnit(TimeUnit timeUnit) {
 	private TimeUnit     timeUnit;
 	private CountDownLatch latch;
 	private boolean somethingFound;
-	
+
 	public ResponseConfiguration() {
 		this(ResponseType.ONE, DEFAULT.getTimeout(), DEFAULT.getTimeUnit());
 	}
-	
+
 	public ResponseConfiguration(ResponseType responseType, long timeout, TimeUnit timeUnit) {
 		super();
 		this.responseType = responseType;
 		this.timeout  = timeout;
 		this.timeUnit = timeUnit;
 	}
-	
+
 	public void latch(ResponseWaiter waiter) throws EventException, InterruptedException {
-		
+
 		if (waiter==null) waiter = new ResponseWaiter.Dont();

```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Introduce a new variable instead of reusing the parameter "waiter". [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1226)